qiskit/extensions/quantum_initializer/ucg.py

```diff
@@ -244,20 +245,20 @@ def _demultiplex_single_uc(self, a, b):
             d = np.flip(d, 0)
             u = np.flip(u, 1)
         d = np.diag(np.sqrt(d))
-        v = d.dot(_ct(u)).dot(_ct(r)).dot(b)
+        v = d.dot(UCG._ct(u)).dot(UCG._ct(r)).dot(b)
         return v, u, r
 
+    @staticmethod
+    def _ct(m):
+        return np.transpose(np.conjugate(m))
 
-def _ct(m):
-    return np.transpose(np.conjugate(m))
+    @staticmethod
+    def _h():
+        return 1 / np.sqrt(2) * np.array([[1, 1], [1, -1]])
 
-
-def _h():
-    return 1 / np.sqrt(2) * np.array([[1, 1], [1, -1]])
-
-
-def _rz(alpha):
-    return np.array([[np.exp(1j * alpha / 2), 0], [0, np.exp(-1j * alpha / 2)]])
+    @staticmethod
+    def _rz(alpha):
+        return np.array([[np.exp(1j * alpha / 2), 0], [0, np.exp(-1j * alpha / 2)]])
```

Comment:
similar for this one, delete the static method and replace uses by `RzGate(alpha).to_matrix()`

Reply:
`qiskit.extensions.standard.rz.RZGate` does not have a `to_matrix` yet. I can create an issue to track that so we can move forward with this PR independently.

Reply:
Submitted in #3304

Reply:
I thought there was a motivation in leaving out some of the `to_matrix` methods in that their textbook definitions would differ from what was returned from the operator by a global phase.

e.g.
```
>>> Operator(RZGate(1.5)).data
array([[1.       +0.j        , 0.       +0.j        ],
       [0.       +0.j        , 0.0707372+0.99749499j]])
>>> _rz(1.5)
array([[0.73168887+0.68163876j, 0.        +0.j        ],
       [0.        +0.j        , 0.73168887-0.68163876j]])
```

Reply:
yeah this is a bigger discussion about global phases and chris has is preparing an RFC for what qiskit convention should look like.. so won't block this PR.

Reply:
Maybe a discussion to have in #3304  @kdk  ?